src/blueapi/service/main.py

```diff
@@ -100,7 +100,7 @@ def restart_handler(handler: BlueskyHandler):
         handler.stop()
         handler.start()
 
-    if handler.initialized:
+    if handler.state.initialized or handler.state.error_message:
```

Comment:
Something can be false for reasons other than being `None`

```suggestion
    if handler.state.initialized or handler.state.error_message is not None:
```

Reply:
Thanks, yes the falsy empty string is still catching me out!